### PR TITLE
Remove providerID field from m3mdata

### PR DESCRIPTION
**Pull request by Xenwar**
Remove providerID field from m3mdata

Reverts https://github.com/metal3-io/cluster-api-provider-metal3/pull/369

### CODE REVIEW

**Review by kashifest — APPROVED** [review]

**Comment by Xenwar** [comment]
on hold until proposal is approved

**Review by smoshiur1237 — COMMENTED** [review]
/lgtm

**Pull request merged**
